## Download FASTQ files from the NCBI-SRA

Computer and OS characteristics:
+ Laptop: System76 Gazelle gaze16-3050
+ OS: Pop!_OS 22.04 LTS x86_64
+ CPU: 11th Gen Intel i7-11800H (16) @ 4.600GHz
+ GPU: Intel TigerLake-H GT1 [UHD Graphics]
+ Memory: 5822MiB / 31952MiB

Internet connection
+ Download speed: ~ 200 Mbps
+ Upload speed: ~ 20 Mbps

We'll download the FASTQ files from the NCBI-SRA using the SRAToolkit. First,
we need to get the SRA identifiers. We'll get the identifiers for the 
AsnicarF_2017 study through the 
[curatedMetagenomicData command line interface](https://github.com/waldronlab/curatedMetagenomicDataTerminal).

If you have administrator permission, the curatedMetagenomicData command line 
can be installed with:

```bash
curl -sSL https://tinyurl.com/cMDTerminal | bash
```

Otherwise, you can install it with the following commands: 

```bash

## Create the bin directory at $HOME
cd $HOME
mkdir bin

Rscript -e 'utils::install.packages("BiocManager", repos = "https://cloud.r-project.org/")'
Rscript -e 'BiocManager::install("waldronlab/curatedMetagenomicDataTerminal")'
Rscript -e 'curatedMetagenomicDataTerminal::install("~/bin/curatedMetagenomicData")'

## Check that it has been installed
which curatedMetagenomicData
#> /home/samuel/bin/curatedMetagenomicData
```

Note: The latest version of R must be installed and available in the PATH.

Get AsnicarF_2017 metadata and get NCBI IDs in a text file:

```bash
curatedMetagenomicData "AsnicarF_2017.+relative_abundance" --metadata > AsnicarF_2017_metadata.tsv
awk 'BEGIN {FS="\t";OFS="\t"} NR==1{for(i=1;i<=NF;i++) f[$i] = i} { print $(f["NCBI_accession"]) }' AsnicarF_2017_metadata.tsv > sra_ids.txt
sed -i -e '/NCBI_accession/d' sra_ids.txt
head sra_ids.txt

#> SRR4052021
#> SRR4052022
#> SRR4052033
#> SRR4052038
#> SRR4052039
#> SRR4052040
#> SRR4052041
#> SRR4052042
#> SRR4052043
#> SRR4052044
```

Now use the SRAToolkit to download the FASTQ files. Refer to the
[SRAToolkit](https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit) documentation for download, installation, and configuration
instructions.

Let's download to current directory (detailed instructions [here](https://github.com/ncbi/sra-tools/wiki/08.-prefetch-and-fasterq-dump)):

```bash
## Configure the SRAToolkit to download to current working diretory (cwd)
vdb-config --prefetch-to-cwd
```

Prefetch the accession numbers

```bash
mkdir fastq_files
cd fastq_files
for i in $(cat ../sra_ids.txt); do prefetch $i; done # ~ 25 min  ~ 200 Mbps 
for i in $(cat ../sra_ids.txt); do fasterq-dump $i; done # ~ 15 min ~ 200 Mbps 

```

The output in the `fastq_files` directory should look like:

```bash
ls -lh | head
#> total 117G
#> drwxrwxr-x 2 samuel samuel 4.0K Jul 18 15:17 SRR4052021
#> -rw-rw-r-- 1 samuel samuel 3.6G Jul 18 15:43 SRR4052021_1.fastq
#> -rw-rw-r-- 1 samuel samuel 3.6G Jul 18 15:43 SRR4052021_2.fastq
#> -rw-rw-r-- 1 samuel samuel  42M Jul 18 15:43 SRR4052021.fastq
#> drwxrwxr-x 2 samuel samuel 4.0K Jul 18 15:18 SRR4052022
#> -rw-rw-r-- 1 samuel samuel 4.2G Jul 18 15:44 SRR4052022_1.fastq
#> -rw-rw-r-- 1 samuel samuel 4.2G Jul 18 15:44 SRR4052022_2.fastq
#> -rw-rw-r-- 1 samuel samuel  45M Jul 18 15:43 SRR4052022.fastq
#> drwxrwxr-x 2 samuel samuel 4.0K Jul 18 15:27 SRR4052023
```

Notice that the total size is over 100 GB.

You get three fastq files per accession. Paired reads (\_1 and \_2) and
unpaired reads (just fastq).


```bash
wc -l SRR4052021*fastq
#>   54803376 SRR4052021_1.fastq
#>   54803376 SRR4052021_2.fastq
#>     607068 SRR4052021.fastq
#>  110213820 total
```


Total time to download data: about 45-60 minutes.





